controllers/validation.js

```diff
@@ -0,0 +1,626 @@
+/*
+
+  ___ ___       ___               _ _       _   ___ ___ 
+ | __/ __|  ___| _ \___ _ _ _ __ (_) |_    /_\ | _ \_ _|
+ | _|\__ \ / -_)  _/ -_) '_| '  \| |  _|  / _ \|  _/| | 
+ |_| |___/ \___|_| \___|_| |_|_|_|_|\__| /_/ \_\_| |___|
+
+*/
+
+//*******************************************************************
+
+'use strict';
+
+//*******************************************************************
+// required modules
+const path = require('path');
+const Validator = require('jsonschema').Validator;
+const include = require('include')(__dirname);
+
+//*******************************************************************
+// other files
+
+const errors = require('./patternErrorMessages.json');
+
+const v = new Validator();
+
+const fileMimes = [
+	'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+	'application/msword',
+	'text/rtf',
+	'application/pdf'
+];
+
+function digitCheck(input, num){
+
+	let valid = true;
+	
+	if (typeof input === 'number'){
+
+		const inputStr = input + '';
+
+		if (!inputStr.match(new RegExp(`^[0-9]{${num}}$`))){
+
+			valid = false;
+
+		}
+
+	}
+
+	return valid;
+
+}
+
+function areaCodeFormat(input){
+
+	return digitCheck(input, 3);
+
+}
+function phoneNumberFormat(input){
+
+	return digitCheck(input, 7);
+
+}
+
+/**
+ * Removes 'instance' from prop field of validation errors. Used to make fields human readable
+ * 
+ * @param {string} prop - Prop field from validation error
+ * @return {string}
+ */
+function removeInstance(prop){
+
+	let fixedProp = '';
+
+	if (prop.indexOf('.') !== -1){
+
+		fixedProp = prop.substring((prop.indexOf('.') + 1), (prop.length));
+
+	}
+
+	return fixedProp;
+
+}
+
+/**
+ * Combines property and argument fields, if property exists, for missing field errors
+ *
+ * @param  {string} property - Upper field to combine
+ * @param  {string} argument - Field where error is.
+ * @return {string} - Concatination of property, '.', and argument
+ */
+function combinePropArgument(property, argument){
+
+	let field;
+	if (property.length > 0){
+
+		field = `${property}.${argument}`;
+
+	}
+	else {
+
+		field = argument;
+
+	}
+
+	return field;
+
+}
+/**
+ * Creates error object which can be read by error message building function
+ * 
+ * @param {string} field - Field where error occured at
+ * @param {string} errorType - Type of error returned
+ * @param {string} expectedFieldType - Type that the field is expected to be
+ * @param {string} enumMessage - Enum message returned by validation
+ * @param {string} dependency - Fields that are a dependeny of field
+ * @param {array} anyOfFields - Array of strings of all field included in anyOf
+ * 
+ * @return Error object
+ */
+function makeErrorObj(field, errorType, expectedFieldType, enumMessage, dependency, anyOfFields){
```

Comment:
Would like to see the field not included rather than passing null. Maybe pass an object with the fields that each error does have.